### DISCUSSION TITLE
Clippy vs cargo action

### DISCUSSION THREAD

**Discussion by jplatte** [actions-rs/meta#32] [Q&A]
Why is there a clippy action separate from the cargo one?

**Comment by svartalf** [comment]
Main advantage of it is being able to create annotations in PRs, as you can see on the screenshot here: https://github.com/actions-rs/clippy-check#rust-clippy-check-action

And whole this functionality (and action itself) is basically a workaround for these two Github issues:
 * https://github.com/actions/toolkit/issues/186
 * https://github.com/actions/toolkit/issues/193

which are preventing to do the same directly in the `cargo` action.

**Reply by jplatte** [reply]
This is for PRs from different repos, right? Since it seems the Github Token being required is one of the main differences. Why could the same workaround the clippy action uses not be done in the cargo action though?
